h2o-core/src/main/java/water/rapids/Merge.java

```diff
@@ -327,26 +329,140 @@ public void map(Chunk[] cs, NewChunk[] ncs) {
     }
     Key<Vec> key = Vec.newKey();
     Vec[] vecs = new Vec(key, Vec.ESPC.rowLayout(key, espc)).makeCons(numColsInResult, 0, doms, types);
-    System.out.println("took: " + (System.nanoTime() - t0) / 1e9);
+    Log.info("took: " + (System.nanoTime() - t0) / 1e9+" seconds.");
 
     System.out.print("Finally stitch together by overwriting dummies ...");
     t0 = System.nanoTime();
     Frame fr = new Frame(names, vecs);
     ChunkStitcher ff = new ChunkStitcher(chunkSizes, chunkLeftMSB, chunkRightMSB, chunkBatch);
     ff.doAll(fr);
-    System.out.println("took: " + (System.nanoTime() - t0) / 1e9);
+    Log.info("took: " + (System.nanoTime() - t0) / 1e9+" seconds");
 
     //Merge.cleanUp();
     return fr;
   }
 
-  private static RadixOrder createIndex(boolean isLeft, Frame fr, int[] cols, int[][] id_maps, int[] ascending) {
-    System.out.println("\nCreating "+(isLeft ? "left" : "right")+" index ...");
+  public static Frame sortOnly(final Frame leftFrame, final int leftCols[], int[][] id_maps, int[] ascendingL) {
```

Comment:
this method is too long and impossible to understand
still some Sys.outs left

Reply:
Let me add more comments and break it up if possible.